mne/filter.py

```diff
@@ -1346,6 +1346,55 @@ def resample(x, up, down, npad=100, axis=-1, window='boxcar', n_jobs=1,
     return y
 
 
+def _resample_stim_channels(stim_data, up, down):
+    """Resample stim channels, carefully.
+
+    Parameters
+    ----------
+    stim_data : 2D array, shape (n_stim_channels, n_samples)
+        Stim channels to resample.
+    up : float
+        Factor to upsample by.
+    down : float
+        Factor to downsample by.
+
+    Returns
+    -------
+    stim_resampled : 2D array, shape (n_stim_channels, n_samples_resampled)
+        The resampled stim channels
+    """
+    stim_data = np.atleast_2d(stim_data)
```

Comment:
Can you make a note of where the equivalent steps are done in the C code? We try to make it easier for the next person to look up code if possible. Did you end up making something equivalent, or just close? Please add a note for that, too (also good to know where we differ, if at all).